stm32cube/stm32h5xx/CMakeLists.txt

```diff
@@ -0,0 +1,119 @@
+# Copyright (c) 2023 STMicroelectronics
+#
+# SPDX-License-Identifier: Apache-2.0
+
+zephyr_library_sources(soc/system_stm32h5xx.c)
+zephyr_library_sources(drivers/src/stm32h5xx_hal.c)
+zephyr_library_sources(drivers/src/stm32h5xx_hal_rcc.c)
+zephyr_library_sources(drivers/src/stm32h5xx_hal_rcc_ex.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_ADC drivers/src/stm32h5xx_hal_adc.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_ADC_EX drivers/src/stm32h5xx_hal_adc_ex.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_CEC drivers/src/stm32h5xx_hal_cec.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_COMP drivers/src/stm32h5xx_hal_comp.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_CORDIC drivers/src/stm32h5xx_hal_cordic.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_CORTEX drivers/src/stm32h5xx_hal_cortex.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_CRC drivers/src/stm32h5xx_hal_crc.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_CRC_EX drivers/src/stm32h5xx_hal_crc_ex.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_CRYP drivers/src/stm32h5xx_hal_cryp.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_CRYP_EX drivers/src/stm32h5xx_hal_cryp_ex.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_DAC drivers/src/stm32h5xx_hal_dac.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_DAC_EX drivers/src/stm32h5xx_hal_dac_ex.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_DCACHE drivers/src/stm32h5xx_hal_dcache.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_DCMI drivers/src/stm32h5xx_hal_dcmi.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_DMA drivers/src/stm32h5xx_hal_dma.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_DMA_EX drivers/src/stm32h5xx_hal_dma_ex.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_DTS drivers/src/stm32h5xx_hal_dts.c)
+if(CONFIG_ETH_STM32_HAL_API_V2)
+	zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_ETH drivers/src/stm32h5xx_hal_eth.c)
+	zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_ETH_EX drivers/src/stm32h5xx_hal_eth_ex.c)
+endif()
```

Comment:
If there's a dependency, better make it explicit, than silently fail if not satisfied

Reply:
Can you try something like :
if (CONFIG_USE_STM32_HAL_ETH)
    if (NOT CONFIG_ETH_STM32_HAL_API_V2)
         message(FATAL_ERROR "command failed with return code: ${ret}")
   endif()
else
    if(CONFIG_ETH_STM32_HAL_API_V2)
	zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_ETH drivers/src/stm32h5xx_hal_eth.c)
	zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_ETH_EX drivers/src/stm32h5xx_hal_eth_ex.c)
    endif()
endif()

Reply:
adding message for the a wrong config: CONFIG_USE_STM32_HAL_ETH but NOT CONFIG_ETH_STM32_HAL_API_V2